### PR TITLE
Typing update

### PR DESCRIPTION
**Pull request by svbrodersen**
Fixes to the check_sample_weight and check_sample_indices also included in the PR.

### CODE REVIEW

File: src/adaXT/base_model.pyx [hunk 1]
```diff
@@ -13,9 +13,52 @@ from .leaf_builder.leaf_builder cimport (LeafBuilderClassification,
...
+        return sample_weight
+
+    def _check_sample_indices(self,
+                              sample_indices: ArrayLike | None
+                              ) -> np.ndarray:
+
+        if sample_indices is None:
+            return np.arange(0, self.X_n_rows, dtype=np.int32)
+        return np.array(sample_indices, dtype=np.int32)
+
```
Comment (NiklasPfister) [inline src/adaXT/base_model.pyx:61]: Is there a reason not to add the same checks as in _check_sample_weight here?

File: src/adaXT/decision_tree/decision_tree.pyx [hunk 2]
```diff
@@ -27,14 +27,14 @@ class refit_object():
             idx: int,
```
Comment (NiklasPfister) [inline src/adaXT/decision_tree/decision_tree.pyx:27]: remove one of the `from numpy.typing import ArrayLike` imports

File: src/adaXT/predict/predict.pyi [hunk 3]
```diff
@@ -36,7 +38,14 @@ class Predict:
         pass
 
     @staticmethod
-    def forest_predict(predictions: np.ndarray, **kwargs):
+    def forest_predict(
+        X_old: np.ndarray,
+        Y_old: np.ndarray,
+        X_new: np.ndarray,
```
Comment (NiklasPfister) [inline src/adaXT/predict/predict.pyi:44]: change these to `cnp.ndarray`?

**Review by NiklasPfister — APPROVED** [review]
Nice! Only had a few very small comments.

File: src/adaXT/base_model.pyx (hunk 1 above)
Comment (svbrodersen) [reply src/adaXT/base_model.pyx:61]: Yes, sample_indices might be a subset of the total number of samples. For example when doing bootstrapping in the random forest, we might only want to have 5 of the total 1000 rows as part of the single decision tree. Even though we skip the check in randomforest this is just an example. Or when someone else is defining it, they might also wish for the functionality. However, the number of dimension check has been added.

File: src/adaXT/predict/predict.pyi (hunk 3 above)
Comment (svbrodersen) [reply src/adaXT/predict/predict.pyi:44]: the pyi file is simply an outwards facing file for the language server. Has no input, but is not cython defined, so can not make use of cnp.

**Review by NiklasPfister — APPROVED** [review]
Seems that this is ready to be merged.

**Pull request merged**